src/ui/server/controllers/insurance/business/company-details/index.ts

```diff
@@ -142,9 +150,57 @@ const postCompaniesHouseSearch = async (req: Request, res: Response) => {
   }
 };
 
+/**
+ * saves and goes back to all sections from company details page unless there are database errors
+ * @param req
+ * @param res
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} redirects to all sections page on success
+ */
+const postCompanyDetailsSaveAndBack = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    const { referenceNumber } = req.params;
+
+    if (!application) {
+      return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+    }
+
+    const { body } = req;
+    // runs companiesHouse validation and api call first for companiesHouse input
+    const response = await companiesHouseSearch(body);
+    let { validationErrors } = response;
+    const { company } = response;
+
+    // run validation on other fields on page
+    validationErrors = companyDetailsValidation(body, validationErrors);
+
+    // body for update containing companies house info and request body
+    const updateBody = {
+      ...body,
+      ...company,
+    };
+
+    // runs save and go back commmand
+    const saveResponse = await mapAndSave.companyDetailsSave(updateBody, application, validationErrors);
```

Comment:
`mapAndSave.companyDetails` is easier to digest IMO (?)

Reply:
done